tests/controls/gridtest.cpp

```diff
@@ -1449,6 +1592,452 @@ TEST_CASE_METHOD(GridTestCase, "Grid::DrawInvalidCell", "[grid][multicell]")
     wxYield();
 }
 
+#define CHECK_ATTR_COUNT(n) CHECK( m_grid->GetCellAttrCount() == n )
+
+TEST_CASE_METHOD(GridTestCase, "Grid::CellAttribute", "[grid][cellattr]")
+{
+    SECTION("Overwrite")
+    {
+        CHECK_ATTR_COUNT( 0 );
+
+        m_grid->SetAttr(0, 0, NULL);
+        CHECK_ATTR_COUNT( 0 );
+
+        SetCellAttr(0, 0);
+        CHECK_ATTR_COUNT( 1 );
+
+        m_grid->SetAttr(0, 0, NULL);
+        CHECK_ATTR_COUNT( 0 );
+
+        SetCellAttr(0, 0);
+        m_grid->SetCellBackgroundColour(0, 1, *wxGREEN);
+        CHECK_ATTR_COUNT( 2 );
+
+        m_grid->SetAttr(0, 1, NULL);
+        m_grid->SetAttr(0, 0, NULL);
+        CHECK_ATTR_COUNT( 0 );
+    }
+
+
+    // Fill the grid with attributes for next sections.
+
+    const int numRows = m_grid->GetNumberRows();
+    const int numCols = m_grid->GetNumberCols();
+
+    for ( int row = 0; row < numRows; ++row )
+        for ( int col = 0; col < numCols; ++col )
+            SetCellAttr(row, col);
+
+    size_t numAttrs = static_cast<size_t>(numRows * numCols);
+
+    CHECK_ATTR_COUNT( numAttrs );
+
+    SECTION("Expanding")
+    {
+        CHECK( !HasCellAttr(numRows, numCols) );
+
+        m_grid->InsertCols();
+        CHECK_ATTR_COUNT( numAttrs );
+        CHECK( !HasCellAttr(0, 0) );
+        CHECK( HasCellAttr(0, numCols) );
+
+        m_grid->InsertRows();
+        CHECK_ATTR_COUNT( numAttrs );
+        CHECK( HasCellAttr(numRows, numCols) );
+    }
+
+    SECTION("Shrinking")
+    {
+        CHECK( HasCellAttr(numRows - 1 , numCols - 1) );
+        CHECK( HasCellAttr(0, numCols - 1) );
+
+        m_grid->DeleteCols();
+        numAttrs -= m_grid->GetNumberRows();
+        CHECK_ATTR_COUNT( numAttrs );
+        CHECK( HasCellAttr(0, 0) );
+        CHECK( !HasCellAttr(0, numCols - 1) );
+
+        m_grid->DeleteRows();
+        numAttrs -= m_grid->GetNumberCols();
+        CHECK_ATTR_COUNT( numAttrs );
+        CHECK( !HasCellAttr(numRows - 1 , numCols - 1) );
+    }
+}
+
+namespace
+{
+
+// Compares two grids, checking for differences with attribute presence and
+// cell sizes.
+class GridMatcher : public Catch::MatcherBase<TestableGrid>
```

Comment:
I wonder if this shouldn't be renamed to `GridAttrMatcher` as I was a bit surprised to see that it didn't actually compare the grid contents.

Reply:
Even after reading that I was first thinking you mean it doesn't compare the properties of attributes. Ah yes, there's such a thing as strings still and cells are even considered empty without one. I'll rename it for now, at least I don't plan on making use of such functionality soon (and other grid tests may already be checking values after row/column editing, I don't know ATM).